sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,217 @@
+# Author: Henry Lin <hlin117@gmail.com>
+
+from __future__ import division
+
+import numpy as np
+from ..externals.six.moves import xrange
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.validation import check_array, check_is_fitted, column_or_1d
+
+
+class KBinsDiscretizer(BaseEstimator, TransformerMixin):
+    """Bins continuous data into k equal width intervals.
+
+    Parameters
+    ----------
+    n_bins : int (default=2)
```

Comment:
If we consider different number of bins per feature, we can't use a `np.array` to store the `bin_edges_`, right? So we would have to use at least a python list to store the bin edges for each feature.

I suppose it doesn't change this code too much. After all, I'm using a for-loop over each column of `bin_edges_` in my current implementation of transform.